lib/_http_outgoing.js

```diff
@@ -109,35 +109,29 @@ OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
 OutgoingMessage.prototype.destroy = function(error) {
-  if (this.socket)
+  if (this.socket) {
+    // If the socket is corked from a write, uncork it before destroying it.
+    if (this.socket._writableState.corked)
+      this.socket.uncork();
```

Comment:
This is not correct. Multiple `cork()` call increase the `corked` counter: https://github.com/nodejs/node/blob/master/lib/_stream_writable.js#L226-L230.

This needs to be a for loop that calls `uncork()` until `corked` is 0.